speech.js

```diff
@@ -82,38 +85,52 @@ class MessageQueue {
 
 class StreamingSynthesisConnection {
   constructor(apiKey, voice, options={}) {
-    this._socket = new WebSocket(URL_STREAMING);
-    this._socket.onerror = (event) => {
-      console.error("WebSocket error:", event);
-      this._inMessages.finish();
-    };
-    this._socket.onmessage = this._onMessage.bind(this);
     this._outMessages = [];
     this._inMessages = new MessageQueue();
     this._return_extras = options.return_extras || false;
+    this._setupWebSocket();
     this._sendMessage({
       'X-API-Key': apiKey,
       'voice': voice,
       'send_extras': options.return_extras || undefined,
       'speed': options.speed || undefined,
       'expressive': options.expressive || undefined,
     });
-    this._socket.onopen = () => {
-      logDebug(`Socket opened.`);
-      this._flushMessages();
-    };
-    this._socket.onclose = (event) => {
-      logDebug(`Socket closed.`);
-      if (event.code !== 1000) {
-        // 1000 is normal closure
-        console.warn(
-          `WebSocket closed unexpectedly with code: ${event.code}, reason: ${event.reason}`
-        );
-      }
-      // TODO(shaper): Consider retrying on reconnect, or clearing queued messages.
-      this._socket = null;
-      this._inMessages.finish();
-    };
+  }
+
+  async _setupWebSocket() {
+    this._socket = await createSimpleWebSocket(URL_STREAMING);
+    this._socket.onerror = this._onError.bind(this);
+    this._socket.onmessage = this._onMessage.bind(this);
+    this._socket.onopen = this._onOpen.bind(this);
+    this._socket.onclose = this._onClose.bind(this);
+  }
+
+  _onError(event) {
+    console.error("WebSocket error:", event);
+    this._inMessages.finish();
+  }
+
+  _onOpen() {
+    logDebug(`Socket opened.`);
+    this._flushMessages();
+  }
+
+  _onClose(event) {
+    logDebug(`Socket closed.`);
+    if (event.code !== 1000) {
```

Comment:
Maybe move this to a constant like you did with the open constant?

Reply:
done